### PR TITLE
FINERACT-1177: Add feature to include external libraries

### PR DESCRIPTION
**Pull request by vidakovic**
## Description

These changes allow us to add external libraries to the Spring Boot classloader without unpacking the Spring Boot uber-jar. Any jar file that is available at "/app/libs" in the Docker container will be available/visible in Fineract during runtime. As part of the Docker image build we download and copy the official Mysql JDBC driver into that directory. Later these mechanics can also be used to load "modules" (like the reporting module that is currently worked on).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,23 +24,22 @@ WORKDIR /fineract
...
-WORKDIR /fineract/target/BOOT-INF/lib
-RUN wget -q https://repo1.maven.org/maven2/mysql/mysql-connector-java/8.0.20/mysql-connector-java-8.0.20.jar
+WORKDIR /fineract/libs
+RUN wget -q https://repo1.maven.org/maven2/mysql/mysql-connector-java/8.0.23/mysql-connector-java-8.0.23.jar
 
 # =========================================
 
-FROM gcr.io/distroless/java:11 as fineract
+#FROM gcr.io/distroless/java:11 as fineract
+FROM openjdk:11 AS fineract
```
Comment (ptuomola) [inline Dockerfile]: Hi - we put in quite a bit of work earlier to move from OpenJDK to distroless for smaller images and better security etc. It's a shame if we now have to undo this and go back to OpenJDK. Would the same changes not work with distroless?

**Review by ptuomola — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (vidakovic) [reply Dockerfile]: Arrgh, my mistake... that was not intended (was just testing something else... unrelated). I'll revert that immediately.

Comment (vidakovic) [reply Dockerfile]: Alright, distroless is in place again. Thanks for the pointer @ptuomola

**Comment by vorburger** [comment]
I'll review this now, and perhaps even quickly try it, and go ahead and ASAP merge it if it LGTM...

**Review by vorburger — APPROVED** [review]
LGTM! I've verified both that...

... manually launching `./gradlew bootJar && java -jar fineract-provider/build/libs/fineract-provider.jar` still works,

... AND given that Travis passed, convinced myself that this "new" way of loading the `mysql-connector-java-8.0.23.jar` clearly still works as well.

I'll try to find the time to adapt https://github.com/vorburger/fineract-pentaho/ to this, hopefully on one of the coming weekends.

It would also be nice to have some documentation for this; I'll see if I can put something together now.

**Pull request merged**